database/gdb/gdb_types.go

```diff
@@ -0,0 +1,11 @@
+package gdb
+
+import "database/sql/driver"
+
+type ScannerForFieldType interface {
+	ScanForFieldType(fieldType string, src any) error
+}
+
+type ValuerForFieldType interface {
+	ValueForFieldType(fieldType string) (driver.Value, error)
+}
```

Comment:
It is suggested make it as simple as possible for usage, not more unnecessary, hidden interfaces.